### PR TITLE
expression: implement vectorized evaluation for `builtinCastDurationAsIntSig`

### PR DESCRIPTION
**Pull request by tangwz**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinCastDurationAsIntSig.
Issue: #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinCastFunc/builtinCastDurationAsIntSig-VecBuiltinFunc-4             2752            412671 ns/op           49989 B/op         2232 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastDurationAsIntSig-NonVecBuiltinFunc-4          2636            426248 ns/op           49989 B/op         2232 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_cast_vec.go [hunk 1]
```diff
@@ -278,11 +278,40 @@ func (b *builtinCastTimeAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result
...
+	result.ResizeInt64(n, false)
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+	d := b.getRetTp().Decimal
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+
+		dur, err := buf.GetDuration(i, d).RoundFrac(types.DefaultFsp)
```
Comment (SunRunAway) [inline expression/builtin_cast_vec.go]: We can use `buf.GoDuration` here, the vectorized version of `buf.GetDuration`.

Comment (qw4990) [inline expression/builtin_cast_vec.go]: How about:
```
var duration Duration
ds := buf.GoDurations()
for i ... {
    ...
    duration.Duration = ds[i]
    d, err := duration.RoundFrac(...)
    ...
}
```

**Comment by qw4990** [comment]
```
[2019-10-22T07:29:16.064Z] integration_test.go:2515:
[2019-10-22T07:29:16.064Z]     result.Check(testkit.Rows("130000"))
[2019-10-22T07:29:16.064Z] /home/jenkins/agent/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:62:
[2019-10-22T07:29:16.064Z]     res.c.Assert(resBuff.String(), check.Equals, needBuff.String(), res.comment)
[2019-10-22T07:29:16.064Z] ... obtained string = "[125959]\n"
[2019-10-22T07:29:16.064Z] ... expected string = "[130000]\n"
[2019-10-22T07:29:16.064Z] ... sql:select cast(a as signed) from t, args:[]
```

Please fix this problem.

**Comment by tangwz** [comment]
@SunRunAway PTAL

**Pull request merged**